hocs/withAuthorization.tsx

```diff
@@ -27,9 +32,15 @@ const withAuthorization = <Props extends object>( //eslint-disable-line
     const { user } = useAuth();
     const id = useRouter().query.id as string;
 
-    if (!id) return null;
+    if (!user) {
+      return <Unauthorized />;
+    }
+
+    if (!id || !user.id) {
+      return <Page editable={false} {...props} />;
+    }
```

Comment:
Isn't this a flaw in the logic?
If I don't log in or sign up, then I don't have an `id` and there is no `user`. But still I am able to see the profile pages.
And then only after this we use the `isAuthorizedToView` function to check if the user can actually view

Reply:
No. It's not a flaw. It prevents the page to always show the unauthorized page if we don't have a user with an id yet. This will only be the case when the page is still loading, so the loading spinner is shown and no information is displayed.

Reply:
hm. unfortunately, you are right. While it prevents the flicker, it also lets unlogged user onto the page (I thought not logged users would have a null user in the context). I think something is off with our context. Do we always reload the user? Otherwise the user should be in the context when we go to our talent page on first render, but it is not. We need to fix that somehow for authorization to work properly.

Reply:
With only looking into the code and not testing it, with our current setup I am expecting that:

1. there is a user like this in our context, if nobody is logged in:
`{ user: { id: '' } }`
2. we don't reload the user on every render. The `AuthProvider` wraps all the components. 
3. there is always a user in the context after signing in or signing up. In the `handleSubmit` of bother pages we set the context user before redirecting.